lm_eval/models/huggingface.py

```diff
@@ -743,19 +742,12 @@ def _encode_pair(
         return context_enc, continuation_enc
 
     def loglikelihood(self, requests: List[Instance]) -> List[Tuple[float, bool]]:
-        new_reqs = []
-        for context, continuation in [req.args for req in requests]:
-            if context == "":
-                # end of text as context
-                context_enc, continuation_enc = (
-                    [self.eot_token_id],
-                    self.tok_encode(continuation),
-                )
-            else:
-                context_enc, continuation_enc = self._encode_pair(context, continuation)
-
-            new_reqs.append(((context, continuation), context_enc, continuation_enc))
-
```

Comment:
While it's helpful to put recurring processes in functions, I think it's more beneficial to keep this as is for clarity to the reader. Having this repeat for 3 functions in my opinion is not really an issue.

Reply:
Will run the linter.

Defer to you, but to be honest I struggled while reading this code and therefore ended up doing the refactor. More than the reuse, breaking it down into smaller helper functions helped me with readability.